modules/runner/src/main/java/org/apache/ignite/internal/app/IgniteImpl.java

```diff
@@ -990,6 +994,37 @@ public CompletableFuture<Ignite> start(Path configPath) {
 
                         return cmgMgr.onJoinReady();
                     }, startupExecutor)
+                    .thenComposeAsync(ignored -> {
+                        CompletableFuture<Void> awaitSelfInLogicalTopologyFuture = new CompletableFuture<>();
+
+                        LogicalTopologyEventListener awaitSelfListener = new LogicalTopologyEventListener() {
+                            @Override
+                            public void onNodeJoined(LogicalNode joinedNode, LogicalTopologySnapshot newTopology) {
+                                if (newTopology.nodes().stream().map(LogicalNode::id).collect(Collectors.toSet()).contains(id())) {
```

Comment:
If `toSet` is imported statically, this will read nicely in plain English: 'collect to set'

Reply:
Sure.